src/freva/_databrowser.py

```diff
@@ -157,111 +401,76 @@ def databrowser(
-        logger.warning("Turning latest off when searching for a specific version.")
-        latest = False
-    core = {True: "latest", False: "files"}[latest]
-    logger.debug("Searching dictionary: %s\n", search_facets)
-    solr_core = SolrFindFiles(core=core)
-    if (facets or all_facets) and not attributes:
-        out = {}
-        search_facets["facet.limit"] = search_facets.pop("facet_limit", -1)
-        for att, values in solr_core._facets(
-            facets=facets or None, latest_version=False, **search_facets
-        ).items():
-            # values come in pairs: (value, count)
-            value_count = len(values) // 2
-            if relevant_only and value_count < 2:
-                continue
-            if count:
-                out[att] = {v: c for v, c in zip(*[iter(values)] * 2)}
-            else:
-                out[att] = values[::2]
-        return out
+    core = {True: "latest", False: "files"}[not multiversion]
+    if (facet or all_facets) and not attributes:
+        warnings.warn(
+            (
+                "The <facet> and <all_facets> arguments will be made "
+                "deprecated from version 2304.0.0 if you want to"
+                "query databrowser facets use the *freva.facet_search* "
+                "method"
+            ),
+            category=DeprecationWarning,
+        )
+        if count:
+            return count_values(
+                facet=facet,
+                time=time,
+                time_select=time_select,
+                multiversion=multiversion,
+                **search_facets,
+            )
+        return facet_search(
+            facet=facet,
+            time=time,
+            time_select=time_select,
+            multiversion=multiversion,
+            **search_facets,
+        )
     if attributes:
-        # select all is none defined but this flag was set
-        results = solr_core._facets(
-            facets=facets or None, latest_version=False, **search_facets
+        warnings.warn(
+            (
+                "The <attribute> argument will be made deprecated in version"
+                " 2304.0.0."
+            ),
+            category=DeprecationWarning,
         )
+        # select all is none defined but this flag was set
+        with warnings.catch_warnings():
+            warnings.filterwarnings(category=PendingDeprecationWarning, action="ignore")
+
+            results = SolrFindFiles(core=core)._facets(
+                facets=facet or None, latest_version=False, **search_facets
+            )
         if relevant_only:
             return (k for k in results if len(results[k]) > 2)
         return (k for k in results)
     if count:
-        return solr_core._retrieve_metadata(**search_facets).num_objects
-    search_results = solr_core._search(
-        batch_size=batch_size,
-        latest_version=latest,
-        uniq_key=uniq_key,
-        **search_facets,
+        warnings.warn(
```

Comment:
given that the users of freva cli are more casual users than the ones developping tools, wouldn't 3 new databrowser related cli commands confuse the user base?

I was looking for the new methods under cli/ but I could not find any. Is there in this version in the TODO list or are they referenced elswhere that I did not see?

Reply:
The cli wouldn't change. The only thing that I want to change is to drop the `--atributes` and `--all-facets` flag.